geode-redis/src/main/java/org/apache/geode/redis/internal/executor/RenameExecutor.java

```diff
@@ -64,6 +69,24 @@ public void executeCommand(Command command, ExecutionHandlerContext context) {
             region.put(newKey, value);
             removeEntry(key, redisDataType, context);
             break;
+          case REDIS_HASH:
+            // TODO this all needs to be done atomically. Add RENAME support to RedisHashCommands
+            RedisHashCommands redisHashCommands =
+                new RedisHashCommandsFunctionExecutor(context.getRegionProvider().getDataRegion());
+            Collection<ByteArrayWrapper> fieldsAndValues = redisHashCommands.hgetall(key);
+            redisHashCommands.del(key);
+            redisHashCommands.del(newKey);
+            redisHashCommands.hset(newKey, new ArrayList<>(fieldsAndValues), false);
+            break;
+          case REDIS_SET:
+            // TODO this all needs to be done atomically. Add RENAME support to RedisSetCommands
```

Comment:
should we put a placeholder story in the icebox for adding RENAME support for RedisHashCommands and RedisSetCommands?

Reply:
good idea

Reply:
I added one